### PR TITLE
Make tooltip hoverable and dismissible

### LINKED ISSUE #75885: [Web]Tooltip widget meets accessibility standard
According to the new internal accessibility rating, the tool tip needs to fulfill two requirements

1. It needs to be dismissible, developer (or by default) can implement an interaction that allows users to dismiss the tooltip without moving the pointer hover or keyboard focus. It also need to be dismissible through esc key.
2. toolbar needs to be hoverable itself. Currently, the tooltip will be dismissed right after the pointer exit the target. We need to make it so the tooltip will stay on when pointer exit the target and hover over the actually tooltip box.

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/75885

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tooltip.dart [hunk 1]
```diff
@@ -190,18 +191,30 @@ class Tooltip extends StatefulWidget {
...
 
+  static final Set<_TooltipState> _openedToolTips = <_TooltipState>{};
+
+  /// Dismiss the most recently shown tooltip.
+  ///
+  /// This method returns true if it successfully dismisses a tooltip. It returns
+  /// false if there is no tooltip shown on the screen.
+  static bool dismissToolTip() {
+    if (_openedToolTips.isNotEmpty) {
+      _openedToolTips.first._hideTooltip(immediately: true);
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/tooltip.dart:212]: Why just the first one?  Why not hide them all?

File: packages/flutter/test/material/tooltip_test.dart [hunk 2]
```diff
@@ -912,6 +912,105 @@ void main() {
     expect(find.text(tooltipText), findsNothing);
   });
 
+  testWidgets('Tooltip text is also when hoverable', (WidgetTester tester) async {
```
Comment (gspencergoog) [inline packages/flutter/test/material/tooltip_test.dart:915]: I think there's a typo in this description, but I'm not sure what you meant to say...

File: packages/flutter/lib/src/material/tooltip.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/material/tooltip.dart:212]: sounds good

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**